### ISSUE TITLE
(codepipeline): Invalid request when setting notification with Chatbot as target

### ISSUE THREAD

**Issue by kornicameister** [aws/aws-cdk#16745]
Cannot create CodeStarNotification from CodePipeline when using Chatbot as target.

### Reproduction Steps

Cannot exactly provide full working code but that snippet should be enough to describe approach I've taken:

```ts
slack_channel = chatbot.SlackChannelConfiguration.from_slack_channel_configuration_arn(
    self,
    'AyDeployment',
    cdk.Arn.format(
        cdk.ArnComponents(
            service='chatbot',
            resource='chat-configuration/slack-channel',
            resource_name='SOME_CHANNEL',
        ),
        self,
    ),
)

# pipeline is cdk.Pipeline
# pipeline.code_pipeline is codepipeline.Pipeline
pipeline.code_pipeline.notify_on(
    'NotifyOnDeployment',
    slack_channel,
    events=[
        cp.PipelineNotificationEvents.PIPELINE_EXECUTION_STARTED,
        cp.PipelineNotificationEvents.PIPELINE_EXECUTION_FAILED,
        cp.PipelineNotificationEvents.PIPELINE_EXECUTION_SUCCEEDED,
    ],
)
```

### What did you expect to happen?

Correctly configured notification.

### What actually happened?

```sh
8:46:50 AM | CREATE_FAILED        | AWS::CodeStarNotifications::NotificationRule | NotifyOnDeploymentE5ED89E4
Resource handler returned message: "Invalid request provided: AWS::CodeStarNotifications::NotificationRule" (Reques
tToken: 9c7badbc-8365-ebb1-f3f9-a055cc3be81e, HandlerErrorCode: InvalidRequest)
```

### Environment

  - **CDK CLI Version  : 1.121.0**
  - **Framework Version: ?**
  - **Node.js Version 14.17.4:** 
  - **OS               : MacOS BigSur**
  - **Python (Version): 3.9.6 **

### Other

- Tried to directly created a notification using `codestarnotifications.NotificationRule` but it failed with same error.
- Tried to set up a channel directly, not import it via ARN, but it failed too
- Managed to create said notification manually in the web console

--- 

This is :bug: Bug Report

**Comment by skinny85** [comment]
Hey @kornicameister,

thanks for opening the issue. Can you show me the resulting template that you get from synthesizing your code? You'll find it in the `cdk.out` directory of your project.

Thanks,
Adam

**Comment by kornicameister** [comment]
Here's the relevant part:

```
    "PipelineNotifyOnDeployment9537DB63": {
      "Type": "AWS::CodeStarNotifications::NotificationRule",
      "Properties": {
        "DetailType": "FULL",
        "EventTypeIds": [
          "codepipeline-pipeline-pipeline-execution-started",
          "codepipeline-pipeline-pipeline-execution-failed",
          "codepipeline-pipeline-pipeline-execution-succeeded"
        ],
        "Name": "PipelineNotifyOnDeployment4369C5CC",
        "Resource": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition"
              },
              ":codepipeline:{region}:}{account}:",
              {
                "Ref": "Pipeline9850B417"
              }
            ]
          ]
        },
        "Targets": [
          {
            "TargetAddress": {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":chatbot:{region}:{account}:chat-configuration/slack-channel/SOME-CHANNEL"
                ]
              ]
            },
            "TargetType": "AWSChatbotSlack"
          }
        ]
      },
      "Metadata": {
        "aws:cdk:path": "VpcLogs/Pipeline/Pipeline/NotifyOnDeployment/Resource"
      }
    },

```

**Comment by skinny85** [comment]
Hmm yeah, that all looks fine. I would double-check the ARN of the Slack Configuration Channel follows that format. If it does, I would make sure the Channel is configured correctly (something like [this problem](https://stackoverflow.com/questions/68147792/aws-chatbot-not-publishing-to-slack)).

**Comment by skinny85** [comment]
Other than that, I'm really at a loss here 😕.

**Comment by kornicameister** [comment]
Would it possible to find request token in cloud trail?

**Comment by espenmeidell** [comment]
I also experienced problems when using `codePipeline.notifyOn` (using TypeScript). Recreating the same notification rule by instantiating a new NotificationRule worked for me (no idea why).

https://docs.aws.amazon.com/cdk/api/latest/python/aws_cdk.aws_codestarnotifications/NotificationRule.html?highlight=notificationrule

**Comment by kornicameister** [comment]
Ok...I found what was wrong and already scheduled an appointment to check my eyes.
What me and @skinny85 have missed is that Chatbot is a global service thus region in arn does not apply to it. 
When I was constructing ARN manually I haven't specified `region=''` which, in such case, defaulted to region of deployment effectively leading to BadRequest. Now that ARN is declared like so:

```
    cdk.Arn.format(
        cdk.ArnComponents(
            service='chatbot',
			region='',
            resource='chat-configuration/slack-channel',
            resource_name='SOME_CHANNEL',
        ),
        self,
    )
```

stack managed to deploy.

What escaped me though is why if failed when I tried to define chatbot directly in code. Other than my mistake somewhere, I can't think of anything.
